File: Python3/0930. Binary Subarrays With Sum.py
class Solution:
    def numSubarraysWithSum(self, nums: List[int], goal: int) -> int:
        sums = collections.defaultdict(int)
        s = 0
        res = 0
        for n in nums:
            sums[s] += 1
            s += n
            res += sums[s - goal] 
        
        return res

File: C++/0930. Binary Subarrays With Sum.cpp
class Solution {
public:
    int numSubarraysWithSum(vector<int>& nums, int goal) {
        unordered_map<int, int> sums;
        int s = 0;
        int res = 0;
        for(auto n:nums) {
            sums[s]++;
            s += n;
            res += sums[s - goal];
        }
        return res;
    }
};
